lib/src/socket.dart

```diff
@@ -609,8 +583,8 @@ class PhoenixSocket {
     _pendingMessages.clear();
   }
 
-  Future<PhoenixSocket?> _delayedReconnect() async {
-    if (_reconnecting) return null;
+  Future<void> _delayedReconnect() async {
+    if (_reconnecting) return;
```

Comment:
[Side note]
This is interesting - if we're already reconnecting, we return early, but it's before the original reconnect attempt completes. Can this be a problem?